### ISSUE TITLE
State::initialize crashes with "KeyError: key 0x0000000001c4cdd0 not found"

### ISSUE THREAD

**Issue by Clemapfel** [Clemapfel/jluna#17]
Not yet reproduced by the devs, but by an anonymous User on ubuntu 20, compiling with G++-11:

Code:
```cpp
jluna::State::initialize();
```

Error:
```cmake
[JULIA][LOG] initialization successfull.
terminate called after throwing an instance of 'jluna::JuliaException'
  what():  [JULIA][EXCEPTION] KeyError: key 0x5845741900000048 not found
Stacktrace:
 [1] getindex
   @ ./dict.jl:481 [inlined]
 [2] free_reference(key::UInt64)
   @ Main.jluna.memory_handler ./none:762
 [3] safe_call(f::Function, args::UInt64)
   @ Main.jluna.exception_handler ./none:517

signal (6): Aborted
in expression starting at none:0
gsignal at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x7f3ff3a10a30)
unknown function (ip: 0x7f3ff3a1c5db)
unknown function (ip: 0x7f3ff3a1b5a8)
__gxx_personality_v0 at /lib/x86_64-linux-gnu/libstdc++.so.6 (unknown line)
unknown function (ip: 0x7f3ff3816493)
_Unwind_Resume at /lib/x86_64-linux-gnu/libgcc_s.so.1 (unknown line)
_ZN5jluna9safe_callIJP11_jl_value_tEEES2_S2_DpT_ at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZN5jluna5State6detail14free_referenceEm at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZN5jluna5Proxy10ProxyValueD1Ev at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZNSt15_Sp_counted_ptrIPN5jluna5Proxy10ProxyValueELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZNSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE10_M_releaseEv at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEaSERKS2_ at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZNSt12__shared_ptrIN5jluna5Proxy10ProxyValueELN9__gnu_cxx12_Lock_policyE2EEaSERKS5_ at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZNSt10shared_ptrIN5jluna5Proxy10ProxyValueEEaSERKS3_ at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZN5jluna5ProxyaSERKS0_ at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZN5jluna4TypeaSEOS0_ at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZN5jluna5State6detail16initialize_typesEv at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZN5jluna5State10initializeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZN5jluna5State10initializeEv at /home/redacted/jluna/build/libjluna.so.0.8.4 (unknown line)
_ZN18MinimalAudioPluginC1Ev at /home/redacted/.vst3/Minimal Plugin Template.vst3/Contents/x86_64-linux/Minimal Plugin Template.so (unknown line)
_Z18createPluginFilterv at /home/redacted/.vst3/Minimal Plugin Template.vst3/Contents/x86_64-linux/Minimal Plugin Template.so (unknown line)
_ZN4juce24createPluginFilterOfTypeENS_14AudioProcessor11WrapperTypeE at /home/redacted/.vst3/Minimal Plugin Template.vst3/Contents/x86_64-linux/Minimal Plugin Template.so (unknown line)
_ZN4juce17JuceVST3ComponentC1EPN9Steinberg3Vst16IHostApplicationE at /home/redacted/.vst3/Minimal Plugin Template.vst3/Contents/x86_64-linux/Minimal Plugin Template.so (unknown line)
_ZN4juceL23createComponentInstanceEPN9Steinberg3Vst16IHostApplicationE at /home/redacted/.vst3/Minimal Plugin Template.vst3/Contents/x86_64-linux/Minimal Plugin Template.so (unknown line)
_ZN4juce17JucePluginFactory14createInstanceEPKcS2_PPv at /home/redacted/.vst3/Minimal Plugin Template.vst3/Contents/x86_64-linux/Minimal Plugin Template.so (unknown line)
unknown function (ip: 0x96cb4d)
unknown function (ip: 0x950220)
unknown function (ip: 0x7bc9e5)
unknown function (ip: 0x79401e)
unknown function (ip: 0x771091)
unknown function (ip: 0x407db3)
unknown function (ip: 0x77e892)
unknown function (ip: 0x7f3ff338e845)
unknown function (ip: 0x7f3ff3381b46)
unknown function (ip: 0x7f3ff3391637)
unknown function (ip: 0x7f3ff33ac3f2)
unknown function (ip: 0x7f3ff33ac866)
unknown function (ip: 0x7f3ff33accec)
unknown function (ip: 0x7f3ff2dd9f78)
unknown function (ip: 0x7f3ff2e0d105)
g_main_context_dispatch at /lib/x86_64-linux-gnu/libglib-2.0.so.0 (unknown line)
unknown function (ip: 0x7f3ff29f03ff)
g_main_context_iteration at /lib/x86_64-linux-gnu/libglib-2.0.so.0 (unknown line)
unknown function (ip: 0x7f3ff33aebf9)
unknown function (ip: 0x7f3ff338ebd1)
unknown function (ip: 0x40d805)
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at ./reaper (unknown line)
Allocations: 796294 (Pool: 795918; Big: 376); GC: 1
Aborted (core dumped)

```

**Comment by Clemapfel** [comment]
Potentially related to the following, reproducible crash:

```cpp
size_t n = -1; // sic
for (size_t i = 0; i < n; ++i)
{
   State::safe_eval("f(x) = 0.5 * x);
   double result = Main[f](double(1));
}
```
Only reproducible in 0.8.4, 0.9.0 restructuring seems to have fixed this

**Comment by Clemapfel** [comment]
Confirmed to be a side-effect of multi-threaded environment. Should be fixed with 0.9, unfixable in 0.8.4
